tests/echo-socket-activated/echo-socket-activated.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strconv"
+
+	"github.com/coreos/rkt/pkg/sys"
+)
+
+const (
+	SD_LISTEN_FDS_START = 3
+)
+
+func sdListenFDs(unsetEnvironment bool) (int, error) {
+	defer func() {
+		if unsetEnvironment {
+			os.Unsetenv("LISTEN_PID")
+			os.Unsetenv("LISTEN_FDS")
+		}
+	}()
+
+	e := os.Getenv("LISTEN_PID")
+	if e == "" {
+		return 0, nil
+	}
+
+	pid, err := strconv.Atoi(e)
+	if err != nil {
+		return -1, err
+	}
+
+	if os.Getpid() != pid {
+		return 0, nil
+	}
+
+	e = os.Getenv("LISTEN_FDS")
+	if e == "" {
+		return 0, nil
+	}
+
+	n, err := strconv.Atoi(e)
+	if err != nil {
+		return -1, err
+	}
+
+	for fd := SD_LISTEN_FDS_START; fd < SD_LISTEN_FDS_START+n; fd++ {
+		if err := sys.CloseOnExec(fd, true); err != nil {
+			return -1, err
+		}
+	}
+
+	return n, nil
+}
+
+func main() {
+	var fd uintptr
+
+	n, err := sdListenFDs(false)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error getting socket-activated FDs: %v\n", err)
+		os.Exit(1)
+	}
+	switch {
+	case n == 1:
+		fd = SD_LISTEN_FDS_START + 0
```

Comment:
Just a remnant from the example in http://0pointer.de/blog/projects/socket-activation.html, not needed in go.